### DISCUSSION TITLE
Can InvenTree handle "raw materials" (e.g. different sizes of cut sheets)?

### DISCUSSION THREAD

**Discussion by chris-thorn** [inventree/InvenTree#3688] [Q&A] [answered]
Hi there,

I've been using InvenTree for a few days and so far I'm in love with it - it's a joy to use, and so flexible with the impot/export functions!

However, there is one thing that I can't quite figure out, and I wanted to know if it's possible with InvenTree: handling what I would call "raw materials".

InvenTree is great for cataloging discrete items like electronic components and fasteners, because each item can have a unique part number. But I would like to use InvenTree to keep track of materials I use for jewellery-making, such as silver sheet and wire, so that I can see what stock I've got at a glance.

An example: I purchase 1mm-thick sterling silver sheet from a supplier. On their website the product code is `CSA 100`. When you order, you enter a length and a width and the material is supplied cut to that size.

What I would like to be able to do is have a part set up in InvenTree that refers to a specific material, and then have various stock items created under that part _but without having to give each item a unique part number_, such as:

**CSA 100** (1mm sterling silver sheet)
* 32mm x 32mm
* 100mm x 75mm
* 50mm x 50mm

**HSA 150** (1.5mm sterling silver round wire)
* 1000mm
* 65mm

**LSA 230** (2.3mm x 1.5mm sterling silver D-shaped wire)
* 150mm

I can see that I can make a template part, and then make variants of that part, but each variant part still requires a unique part number.

One way I can see to store different size/shape materials in InvenTree is to create each item as a part with the Part Name containing the dimensions, e.g.:

* CSA 100 32x32
* CSA 100 100x75
* CSA 100 50x50
* HSA 150 1000
* HSA 150 65
* LSA 230 150

... but this is a lot of work and will create lots of individual parts that will require filtering to be able to see what stock I have of a particular material.

Another way I've found is just to add the dimensions in the "Packaging" field of each stock item for a part:

![image](https://user-images.githubusercontent.com/3541291/190851555-55cca0e9-3c44-4972-9a78-234b3ac4519c.png)

... but then the dimensions of the stock item are not parametric - it's just text in a field.

Does anyone know if what I would like to achieve is possible?

Thanks for taking the time to read this!

Chris

**Comment by miggland** [comment, marked as answer]
What you could do straight away is define a different unit for stock-keeping. The value is discrete, but doesn't have to mean "pieces". I have some stuff in stock where I use different units:
- tubing, unit "m" - eg 30 m in stock.
- insulation sheet, unit "cm2" - eg 3200 cm2 in stock.

You could use something like the second variant as well, and note the actual sizes in notes/packaging for each stock item. This saves you making different parts for each sheet at least. Use a unit which will always be integer, maybe mm2 in your case. It won't give you every detail about sheet sizes on first view, of course.

**Reply by chris-thorn** [reply]
That's an interesting idea. I did consider using the area of pieces of sheet, but decided it's not particularly helpful for my needs. However, when there is only one dimension you're concerned with (e.g. length) then changing the stock item units to mm, say, seems to work as I'd like:

![image](https://user-images.githubusercontent.com/3541291/190866315-82d8a662-7db1-40c5-ae22-e48c7c4e892b.png)

The 1065mm total length is composed of a 1000mm length and a 65mm length:

![image](https://user-images.githubusercontent.com/3541291/190866330-8862efca-49b2-44c7-97eb-778bde2d8798.png)

**Reply by matmair** [reply]
@chris-thorn do you need further assistance or can we close this as answered?

**Reply by SchrodingersGat** [reply]
@chris-thorn with the implementation of [supplier part pack size](https://github.com/inventree/InvenTree/pull/3644) this might be enough for you?

**Reply by chris-thorn** [reply]
> @chris-thorn do you need further assistance or can we close this as answered?

For the moment, being able to change the stock unit to "mm" and storing each length of a specific type of material works well, so please feel free to close.

Being able to have configurable parameters under which a single part can be stocked, such as length and width, is still a feature I would find useful, but for the amount of stuff I need to stock it's by no means an urgent requirement, given everything else you have planned!

Thanks very much.

Chris

**Reply by chris-thorn** [reply]
> @chris-thorn with the implementation of [supplier part pack size](https://github.com/inventree/InvenTree/pull/3644) this might be enough for you?

That's interesting! I will be sure to check this out when I can, because I do tend to buy different sized containers of the same product from time to time. Thanks!
